lib/ClangImporter/ImportDecl.cpp

```diff
@@ -3391,6 +3394,14 @@ namespace {
         func->setAccess(AccessLevel::Public);
       }
 
+      if (!isa<clang::CXXConstructorDecl>(decl) && !importedType) {
```

Comment:
It seems like `!importedType` should be a sufficient condition here. Why try importing it twice?